plugin/python_dependency_usage/python_dependency_usage.go

```diff
@@ -0,0 +1,191 @@
+package python_dependency_usage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/safedep/code/core"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type DependencyUsagePlugin struct{}
+
+// Verify contract
+var _ core.TreePlugin = (*DependencyUsagePlugin)(nil)
+
+func (p *DependencyUsagePlugin) Name() string {
+	return "DependencyUsagePlugin"
+}
+
+func (p *DependencyUsagePlugin) AnalyzeTree(ctx context.Context, tree core.ParseTree) error {
+	lang, err := tree.Language()
+	if err != nil {
+		return fmt.Errorf("failed to get language: %w", err)
+	}
+
+	if lang.Meta().Name != "python" {
```

Comment:
Hmm. String comparison is not good. We should define constants in `core/lang.go` such as:

```
LANGUAGE_PYTHON
```

Use constants for any comparison that drives the algorithm / implementation logic. We should not do string comparisons as it leads to high error possibility. For example, someone might change `python` to `Python` in the language implementation and we don't have language enforced type checking to identify such errors.

Reply:
This is implemented in PR #5 You need to update the code to use `Code` instead of `Name`

Reply:
Added a `SupportedLanguages() []LanguageCode` constraint for every plugin for this purpose and moved language verification to executor